src/App.jsx

```diff
@@ -58,63 +67,84 @@ const CustomPagingInfoView = ({ totalResults }) => {
 
 export default function App() {
   useSearchFocusHotkey();
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const openForm = () => {
+    setModalOpen(true);
+  };
+  const closeForm = () => {
+    setModalOpen(false);
+  };
 
   return (
-    <SearchProvider config={config}>
-      <WithSearch mapContextToProps={({ wasSearched }) => ({ wasSearched })}>
-        {({ wasSearched }) => {
-          return (
-            <div className="App btc-search">
-              <ErrorBoundary>
-                <div className="header">
-                  <img src={logo} className="logo" alt="bitcoin logo" />
-                  <p className="description">Technical Bitcoin Search</p>
-                </div>
-                <Layout
-                  header={
-                    <SearchBox
-                      autocompleteMinimumCharacters={3}
-                      // autocompleteResults={{
-                      //   linkTarget: "_blank",
-                      //   sectionTitle: "Suggested Queries",
-                      //   titleField: "title",
-                      //   urlField: "nps_link",
-                      //   shouldTrackClickThrough: true,
-                      //   clickThroughTags: ["test"],
-                      // }}
-                      autocompleteSuggestions={true}
-                      debounceLength={0}
-                    />
-                  }
-                  sideContent={
-                    <div>
-                      {wasSearched}
-                      {getFacetFields().map((field) => (
-                        <Facet
-                          key={field}
-                          field={field}
-                          isFilterable={getFacetWithSearch().includes(field)}
-                          label={field}
-                          view={CustomMultiCheckboxFacet}
-                        />
-                      ))}
-                    </div>
-                  }
-                  bodyContent={<CustomResults shouldTrackClickThrough={true} />}
-                  bodyHeader={
-                    <>
-                      {wasSearched && (
-                        <PagingInfo view={CustomPagingInfoView} />
-                      )}
-                    </>
-                  }
-                  bodyFooter={<Paging />}
-                />
-              </ErrorBoundary>
-            </div>
-          );
-        }}
-      </WithSearch>
-    </SearchProvider>
+    <ChakraProvider>
```

Comment:
Can you give me some more details about the need for chakra? Not opposed, just looking to better understand.

Reply:
Good point. Well, chakra is a modular, accessible component library for react. It's quite popular in the community.
It takes care of accessibility and includes web best practices out of the box. 

For instance, building the modal component from scratch I would have to take into account the overlay accessibility, hiding it from screen readers ensuring it's non-focusable, then focus input in the form, prevent scroll on modal open, etc. There are a lot of edge cases and chakraui simplifies that with a highly customizable approach.

And is solely an opt-in UI library, you can build components without chakra or include it in some parts of your components. 
For future advantages, tooltips, dropdowns, and even truncating text like the body of search results can benefit from it.